rosbag2/src/rosbag2/rosbag2_node.cpp

```diff
@@ -61,12 +62,69 @@ std::shared_ptr<GenericSubscription> Rosbag2Node::create_generic_subscription(
 
   return subscription;
 }
+
+std::shared_ptr<rcutils_string_map_t> get_initialized_string_map()
+{
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  auto substitutions_map = new rcutils_string_map_t;
+  *substitutions_map = rcutils_get_zero_initialized_string_map();
+  rcutils_ret_t map_init = rcutils_string_map_init(substitutions_map, 0, allocator);
+  if (map_init != RCUTILS_RET_OK) {
+    ROSBAG2_LOG_ERROR("Failed to initialize string map within rcutils.");
+    return std::shared_ptr<rcutils_string_map_t>();
+  }
+  return std::shared_ptr<rcutils_string_map_t>(substitutions_map,
+           [](rcutils_string_map_t * map) {
+             rcl_ret_t cleanup = rcutils_string_map_fini(map);
+             delete map;
+             if (cleanup != RCL_RET_OK) {
+               ROSBAG2_LOG_ERROR("Failed to deallocate string map when expanding topic.");
+             }
+           });
+}
+
+std::string Rosbag2Node::expand_topic_name(std::string topic_name)
```

Comment:
should this function take a `const std::string &`

Reply:
Absolut.